src/devices/src/virtio/net/persist.rs

```diff
@@ -83,7 +83,10 @@ impl Persist<'_> for Net {
         let mut net = Net::new_with_tap(
             state.id.clone(),
             state.tap_if_name.clone(),
-            None,
+            Some(MacAddr::from_bytes_unchecked(
```

Comment:
With this change, the initial bug should also be fixed?

Is there an integration test that we need to adjust?  Maybe test_get_full_config_after_restoring_snapshot

Reply:
Doesn't seem like it needs any change/adjustment.